e2e/nx-plugin-e2e/tests/nx-plugin.e2e.test.ts

```diff
@@ -0,0 +1,256 @@
+    const cwd = join(baseDir, 'configuration-already-configured');
+    registerPluginInWorkspace(
+      tree,
+      join(relativePathToCwd(cwd), 'dist/packages/nx-plugin'),
+    );
+    const { root } = readProjectConfiguration(tree, project);
+    generateCodePushupConfig(tree, root);
+    await materializeTree(tree, cwd);
+
+    const { code, projectJson } = await nxShowProjectJson(cwd, project);
+
+    expect(code).toBe(0);
+
+    expect(projectJson.targets).toStrictEqual(
+      expect.not.objectContaining({
+        ['code-pushup--configuration']: expect.any(Object),
+      }),
+    );
+    expect(projectJson).toMatchSnapshot();
+  });
+
+  it('should add executor target dynamically if the project is configured', async () => {
+    const cwd = join(baseDir, 'add-executor-dynamically');
+    registerPluginInWorkspace(
+      tree,
+      join(relativePathToCwd(cwd), 'dist/packages/nx-plugin'),
+    );
+    const { root } = readProjectConfiguration(tree, project);
+    generateCodePushupConfig(tree, root);
+    await materializeTree(tree, cwd);
+
+    const { code, projectJson } = await nxShowProjectJson(cwd, project);
+    expect(code).toBe(0);
+
+    expect(projectJson.targets).toStrictEqual({
+      ['code-pushup']: {
+        configurations: {},
+        executor: `@code-pushup/nx-plugin:autorun`,
+        options: {},
+      },
+    });
+
+    expect(projectJson).toMatchSnapshot();
+  });
+
+  it('should execute dynamic executor target', async () => {
+    const cwd = join(baseDir, 'execute-dynamic-executor');
+    const pathRelativeToPackage = relative(join(cwd, 'libs', project), cwd);
+    registerPluginInWorkspace(tree, {
+      plugin: join(relativePathToCwd(cwd), 'dist/packages/nx-plugin'),
+      options: {
+        bin: join(relativePathToCwd(cwd), 'dist/packages/nx-plugin'),
+      },
+    });
+    const { root } = readProjectConfiguration(tree, project);
+    generateCodePushupConfig(tree, root, {
+      fileImports: `import type {CoreConfig} from "${join(
+        relativePathToCwd(cwd),
+        pathRelativeToPackage,
+        'dist/packages/models',
+      )}";`,
+      plugins: [
+        {
+          fileImports: `import jsPackagesPlugin from "${join(
+            relativePathToCwd(cwd),
+            pathRelativeToPackage,
+            'dist/packages/plugin-js-packages',
+          )}";`,
+          // @TODO improve formatObjectToJsString to get rid of the "`" hack
+          codeStrings: 'await jsPackagesPlugin({packageManager: `npm`})',
+        },
+      ],
+    });
+    await materializeTree(tree, cwd);
+
+    const { stdout, stderr } = await executeProcess({
+      command: 'npx',
+      args: ['nx', 'run', `${project}:code-pushup -- --dryRun`],
+      cwd,
+    });
+
+    const cleanStderr = removeColorCodes(stderr);
+    // @TODO create test environment for working plugin. This here misses package.jock to execute correctly
```

Comment:
I assume you mean one of these? :grin: 

```suggestion
    // @TODO create test environment for working plugin. This here misses package.json to execute correctly
```

```suggestion
    // @TODO create test environment for working plugin. This here misses package-lock.json to execute correctly
```